### PR TITLE
[zh-cn] update regular expression quantifier

### PR DESCRIPTION
**Pull request by jasonren0403**
### Description

Add reference translation and update guide translation

### Related issues and pull requests

part of #13254

### CODE REVIEW

File: files/zh-cn/web/javascript/guide/regular_expressions/quantifiers/index.md [hunk 1]
```diff
@@ -48,12 +52,10 @@ slug: Web/JavaScript/Guide/Regular_expressions/Quantifiers
...
       <td>
         <p>
-          将前面的项“x”匹配 0 或 1 次。例如，/e?le?/匹配 angel 中的 el 和 angle
-          中的 le。
+          将前面的项“x”匹配 0 或 1 次。例如，/e?le?/匹配“angel”中的“el”和“angle”中的“le”。
         </p>
         <p>
-          如果立即在任何量词*、+、?或{}之后使用，则使量词是非贪婪的
-          (匹配最小次数)，而不是默认的贪婪的 (匹配最大次数)。
+          如果立即在任何 <code>*</code>、<code>+</code>、<code>?</code> 或 <coe>{}</code> 量词之后使用，则使量词变为非贪婪匹配（匹配最小次数），而不是默认的贪婪匹配（匹配最大次数）。
```
[the remark this answers is not in the document]
Comment (jasonren0403) [reply files/zh-cn/web/javascript/guide/regular_expressions/quantifiers/index.md:58]:
```suggestion
          如果立即在任何 <code>*</code>、<code>+</code>、<code>?</code> 或 <code>{}</code> 量词之后使用，则使量词变为非贪婪匹配（匹配最小次数），而不是默认的贪婪匹配（匹配最大次数）。
```

File: files/zh-cn/web/javascript/reference/regular_expressions/quantifier/index.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+## 浏览器兼容性
+
+{{Compat}}
+
+## 参见
+
+- [量词](/zh-CN/docs/Web/JavaScript/Guide/Regular_expressions/Quantifiers)指南
+- [正则表达式](/zh-CN/docs/Web/JavaScript/Reference/Regular_expressions)
+- [析取符：`|`](/zh-CN/docs/Web/JavaScript/Reference/Regular_expressions/Disjunction)
+- [字符类：`[...]`、`[^...]`](/zh-CN/docs/Web/JavaScript/Reference/Regular_expressions/Character_class)
```
[the remark this answers is not in the document]
Comment (jasonren0403) [reply files/zh-cn/web/javascript/reference/regular_expressions/quantifier/index.md:182]:
```suggestion
- [字符类：`[...]`、`[^...]`](/zh-CN/docs/Web/JavaScript/Reference/Regular_expressions/Character_class)
```

[the remark this answers is not in the document]
Comment (yin1999) [reply files/zh-cn/web/javascript/reference/regular_expressions/quantifier/index.md:182]: EOF

```suggestion
- [字符类：`[...]`、`[^...]`](/zh-CN/docs/Web/JavaScript/Reference/Regular_expressions/Character_class)

```

File: files/zh-cn/web/javascript/reference/regular_expressions/quantifier/index.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+## 浏览器兼容性
+
+{{Compat}}
+
+## 参见
+
+- [量词](/zh-CN/docs/Web/JavaScript/Guide/Regular_expressions/Quantifiers)指南
+- [正则表达式](/zh-CN/docs/Web/JavaScript/Reference/Regular_expressions)
+- [析取符：`|`](/zh-CN/docs/Web/JavaScript/Reference/Regular_expressions/Disjunction)
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/regular_expressions/quantifier/index.md:181]: https://github.com/mdn/translated-content/issues/13254#issuecomment-1566065033

"选择" or "选择表达式"?

```suggestion
- [选择：`|`](/zh-CN/docs/Web/JavaScript/Reference/Regular_expressions/Disjunction)
```

Comment (jasonren0403) [reply files/zh-cn/web/javascript/reference/regular_expressions/quantifier/index.md:181]:
```suggestion
- [选择表达式：`|`](/zh-CN/docs/Web/JavaScript/Reference/Regular_expressions/Disjunction)
```

File: files/zh-cn/web/javascript/guide/regular_expressions/quantifiers/index.md [hunk 4]
```diff
@@ -123,37 +121,37 @@ slug: Web/JavaScript/Guide/Regular_expressions/Quantifiers
...
 ```js
-var wordEndingWithAs = /\w+a+/;
-var delicateMessage = "This is Spartaaaaaaa";
+const wordEndingWithAs = /\w+a+\b/;
+const delicateMessage = "This is Spartaaaaaaa";
 
 console.table(delicateMessage.match(wordEndingWithAs)); // [ "Spartaaaaaaa" ]
 ```
 
 ### 计算字符集
```
[the remark this answers is not in the document]
Comment (jasonren0403) [reply files/zh-cn/web/javascript/guide/regular_expressions/quantifiers/index.md:130]:
```suggestion
### 统计单词
```

File: files/zh-cn/web/javascript/guide/regular_expressions/quantifiers/index.md [hunk 5]
```diff
@@ -11,6 +13,8 @@ slug: Web/JavaScript/Guide/Regular_expressions/Quantifiers
 
 ## 类型
 
+> **备注：** 在下文中，*项*不仅指单个字符，还包括[字符类](/zh-CN/docs/Web/JavaScript/Guide/Regular_expressions/Character_classes)、[组和反向引用](/zh-CN/docs/Web/JavaScript/Guide/Regular_expressions/Groups_and_backreferences) 。
```
Comment (jasonren0403) [inline files/zh-cn/web/javascript/guide/regular_expressions/quantifiers/index.md:16]:
```suggestion
> **备注：** 在下文中，*项*不仅指单个字符，还包括[字符类](/zh-CN/docs/Web/JavaScript/Guide/Regular_expressions/Character_classes)、[组和反向引用](/zh-CN/docs/Web/JavaScript/Guide/Regular_expressions/Groups_and_backreferences)。
```

File: files/zh-cn/web/javascript/guide/regular_expressions/quantifiers/index.md [hunk 6]
```diff
@@ -63,8 +65,8 @@ slug: Web/JavaScript/Guide/Regular_expressions/Quantifiers
       </td>
       <td>
         <p>
-          其中“n”是一个正整数，与前一项“x”的 n 次匹配。例如，<code>/a{2}/ </code
-          >不匹配“candy”中的“a”，但它匹配“caandy”中的所有“a”，以及“caaandy”中的前两个“a”。
+          其中“n”是一个正整数，与前一项“x”的 n 次匹配。例如，<code>/a{2}/</code
```
[the remark this answers is not in the document]
Comment (yin1999) [reply files/zh-cn/web/javascript/guide/regular_expressions/quantifiers/index.md:68]: > How should we update the en-US content? @yin1999

Yes, please :+1:

**Comment by jasonren0403** [comment]
raised issue https://github.com/mdn/content/issues/33486, waiting for possible content change idea.

**Review by yin1999 — APPROVED** [review]

**Pull request merged**